### ISSUE TITLE
VisualScript compilation error on target=release

### ISSUE THREAD

**Issue by Calinou** [godotengine/godot#6291]
Host operating system is Arch Linux, compiling Godot Git commit https://github.com/godotengine/godot/commit/9c6175db11ff72603ae58789a462b33ec1e910f8 for macOS without tools, using Xcode SDK 7.2. It also failed on a few previous commits with similar errors.

iOS compilation also has errors like this.

```
Compiling ==> modules/visual_script/visual_script_nodes.cpp
modules/visual_script/visual_script_nodes.cpp:2258:51: error: non-friend class member 'instance' cannot have a qualified name
VisualScriptNodeInstance* VisualScriptCustomNode::instance(VisualScriptInstance* p_instance) {
                          ~~~~~~~~~~~~~~~~~~~~~~~~^
modules/visual_script/visual_script_nodes.cpp:2281:30: error: non-friend class member '_bind_methods' cannot have a qualified name
void VisualScriptCustomNode::_bind_methods() {
     ~~~~~~~~~~~~~~~~~~~~~~~~^
modules/visual_script/visual_script_nodes.cpp:2317:25: error: non-friend class member 'VisualScriptCustomNode' cannot have a qualified name
VisualScriptCustomNode::VisualScriptCustomNode() {
~~~~~~~~~~~~~~~~~~~~~~~~^
modules/visual_script/visual_script_nodes.cpp:2326:26: error: non-friend class member 'get_output_sequence_port_count' cannot have a qualified name
int VisualScriptSubCall::get_output_sequence_port_count() const {
    ~~~~~~~~~~~~~~~~~~~~~^
modules/visual_script/visual_script_nodes.cpp:2331:27: error: non-friend class member 'has_input_sequence_port' cannot have a qualified name
bool VisualScriptSubCall::has_input_sequence_port() const{
     ~~~~~~~~~~~~~~~~~~~~~^
modules/visual_script/visual_script_nodes.cpp:2336:26: error: non-friend class member 'get_input_value_port_count' cannot have a qualified name
int VisualScriptSubCall::get_input_value_port_count() const{
    ~~~~~~~~~~~~~~~~~~~~~^
modules/visual_script/visual_script_nodes.cpp:2348:26: error: non-friend class member 'get_output_value_port_count' cannot have a qualified name
int VisualScriptSubCall::get_output_value_port_count() const{
    ~~~~~~~~~~~~~~~~~~~~~^
modules/visual_script/visual_script_nodes.cpp:2353:29: error: non-friend class member 'get_output_sequence_port_text' cannot have a qualified name
String VisualScriptSubCall::get_output_sequence_port_text(int p_port) const {
       ~~~~~~~~~~~~~~~~~~~~~^
modules/visual_script/visual_script_nodes.cpp:2358:35: error: non-friend class member 'get_input_value_port_info' cannot have a qualified name
PropertyInfo VisualScriptSubCall::get_input_value_port_info(int p_idx) const{
             ~~~~~~~~~~~~~~~~~~~~~^
modules/visual_script/visual_script_nodes.cpp:2370:35: error: non-friend class member 'get_output_value_port_info' cannot have a qualified name
PropertyInfo VisualScriptSubCall::get_output_value_port_info(int p_idx) const{
             ~~~~~~~~~~~~~~~~~~~~~^
modules/visual_script/visual_script_nodes.cpp:2381:29: error: non-friend class member 'get_caption' cannot have a qualified name
String VisualScriptSubCall::get_caption() const {
       ~~~~~~~~~~~~~~~~~~~~~^
modules/visual_script/visual_script_nodes.cpp:2387:29: error: non-friend class member 'get_text' cannot have a qualified name
String VisualScriptSubCall::get_text() const {
       ~~~~~~~~~~~~~~~~~~~~~^
modules/visual_script/visual_script_nodes.cpp:2400:29: error: non-friend class member 'get_category' cannot have a qualified name
String VisualScriptSubCall::get_category() const {
       ~~~~~~~~~~~~~~~~~~~~~^
modules/visual_script/visual_script_nodes.cpp:2431:48: error: non-friend class member 'instance' cannot have a qualified name
VisualScriptNodeInstance* VisualScriptSubCall::instance(VisualScriptInstance* p_instance) {
                          ~~~~~~~~~~~~~~~~~~~~~^
modules/visual_script/visual_script_nodes.cpp:2447:27: error: non-friend class member '_bind_methods' cannot have a qualified name
void VisualScriptSubCall::_bind_methods() {
     ~~~~~~~~~~~~~~~~~~~~~^
modules/visual_script/visual_script_nodes.cpp:2453:22: error: non-friend class member 'VisualScriptSubCall' cannot have a qualified name
VisualScriptSubCall::VisualScriptSubCall() {
~~~~~~~~~~~~~~~~~~~~~^
modules/visual_script/visual_script_nodes.cpp:2462:26: error: non-friend class member 'get_output_sequence_port_count' cannot have a qualified name
int VisualScriptComment::get_output_sequence_port_count() const {
    ~~~~~~~~~~~~~~~~~~~~~^
modules/visual_script/visual_script_nodes.cpp:2467:27: error: non-friend class member 'has_input_sequence_port' cannot have a qualified name
bool VisualScriptComment::has_input_sequence_port() const{
     ~~~~~~~~~~~~~~~~~~~~~^
modules/visual_script/visual_script_nodes.cpp:2472:26: error: non-friend class member 'get_input_value_port_count' cannot have a qualified name
int VisualScriptComment::get_input_value_port_count() const{
    ~~~~~~~~~~~~~~~~~~~~~^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
scons: *** [modules/visual_script/visual_script_nodes.osx.opt.32.o] Error 1
1 warning generated.
scons: building terminated because of errors.
```

**Comment by vnen** [comment]
This is a problem for any platform with `target=release`.

**Comment by akien-mga** [comment]
Fixed by #6352.
